Sources/MediaNavigationBar.swift

```diff
@@ -0,0 +1,128 @@
+/*****************************************************************************
+ * MediaNavigationBar.swift
+ *
+ * Copyright © 2019 VLC authors and VideoLAN
+ *
+ * Authors: Robert Gordon <robwaynegordon # gmail.com>
+ *
+ *
+ * Refer to the COPYING file of the official project for license.
+ *****************************************************************************/
+
+@objc (VLCMediaNavigationBarDelegate)
+protocol MediaNavigationBarDelegate {
+    func didMinimizePlayback(_ navigationBar: MediaNavigationBar)
+    func didToggleChromeCast(_ navigationBar: MediaNavigationBar)
+    func didToggleAirPlay(_ navigationBar: MediaNavigationBar)
+}
+
+@objc (VLCMediaNavigationBar)
+@objcMembers class MediaNavigationBar: UIView {
+    
+    // MARK: Instance Variables
+    weak var delegate: MediaNavigationBarDelegate?
+    
+    var hideTitleLabel: Bool {
+        didSet {
+            mediaTitleTextLabel.isHidden = hideTitleLabel
+        }
+    }
+    
+    lazy var minimizePlaybackButton: UIButton = {
+        var minButton = UIButton(type: .system)
+        minButton.addTarget(self, action: #selector(minimizePlayback), for: .touchUpInside)
+        minButton.setImage(UIImage(named: "iconChevron"), for: .normal)
+        minButton.tintColor = .white
+        minButton.translatesAutoresizingMaskIntoConstraints = false
+        return minButton
+    }()
+    
+    private var mediaTitleTextLabel: UILabel = {
+        var label = UILabel()
+        label.numberOfLines = 1
+        label.textColor = .white
+        label.font = UIFont(name: "SFProDisplay-Medium", size: 17)
+        label.text = NSLocalizedString("TITLE", comment: "Video Title")
+        label.translatesAutoresizingMaskIntoConstraints = false
+        return label
+    }()
+    
+    lazy var chromeCastButton: UIButton = {
+        var chromeButton = UIButton(type: .system)
+        chromeButton.addTarget(self, action: #selector(toggleChromeCast), for: .touchUpInside)
+        chromeButton.setImage(UIImage(named: "renderer"), for: .normal)
+        chromeButton.tintColor = .white
+        chromeButton.translatesAutoresizingMaskIntoConstraints = false
+        return chromeButton
+    }()
+    
+    lazy var airPlayButton: UIButton = {
+        var airPlayButton = UIButton(type: .system)
+        airPlayButton.addTarget(self, action: #selector(toggleAirPlay), for: .touchUpInside)
+        airPlayButton.setImage(UIImage(named: "iconAirplay"), for: .normal)
+        airPlayButton.tintColor = .white
+        airPlayButton.translatesAutoresizingMaskIntoConstraints = false
+        return airPlayButton
```

Comment:
I believe there is a system component called AVRoutepickerview that needs to be used here instead https://developer.apple.com/documentation/avkit/avroutepickerview

but I'm not sure if that will work with our openGL player

Reply:
This is only available in xcode 11, should I just implement it without fallback?

Reply:
ugh fuuuu okay yeah I would implement it with fallback and just if #available or something like that. That way we can just drop the legacy fun when we drop ios 9 and 10 one day

Reply:
"ugh fuuuu" hahahah 😂okay done